### PR TITLE
feature: add skip upgrade cilium CNI flag to API

### PR DESCRIPTION
**Pull request by chrisdoherty4**
This adds a flag - `skipUpgrade` - to the Cilium configuration in our v1alpha1 API that will enable self managed CNIs.

### CODE REVIEW

File: pkg/api/v1alpha1/cluster_types.go [hunk 1]
```diff
@@ -678,6 +678,12 @@ type CNIConfig struct {
 type CiliumConfig struct {
 	// PolicyEnforcementMode determines communication allowed between pods. Accepted values are default, always, never.
 	PolicyEnforcementMode CiliumPolicyEnforcementMode `json:"policyEnforcementMode,omitempty"`
+
+	// SkipUpgrade indicicates that Cilium maintenance should be skipped during upgrades. This can
+	// be used when operators wish to self manage the Cilium installation.
+	// +kubebuilder:default=false
+	// +optional
+	SkipUpgrade *bool `json:"skipUpgrade,omitempty"`
```
Comment (jiayiwang7) [inline pkg/api/v1alpha1/cluster_types.go:686]: why do we need a pointer here?

**Review by jiayiwang7 — COMMENTED** [review]
can we run `make release-manifests` to auto update the crd and eksa-components.yaml?

File: pkg/api/v1alpha1/cluster_types.go (hunk 1 above)
Comment (chrisdoherty4) [reply pkg/api/v1alpha1/cluster_types.go:686]: To align with Kubernetes API conventions.

https://github.com/zecke/Kubernetes/blob/master/docs/devel/api-conventions.md#optional-vs-required

**Pull request merged**